keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/proposal/ProposalDistribution.java

```diff
@@ -42,7 +43,7 @@ default double logProbAtFromGivenTo(Proposal proposal) {
     default double logProbAtToGivenFrom(Proposal proposal) {
         double sumLogProb = 0.0;
         for (Vertex v : proposal.getVerticesWithProposal()) {
-            sumLogProb += logProb(v, proposal.getProposalTo(v), proposal.getProposalFrom(v));
+            sumLogProb += logProb((Probabilistic) v, proposal.getProposalTo(v), proposal.getProposalFrom(v));
```

Comment:
We should discuss, because I'm not sure I understand all the "Proposal" stuff well enough, but should a `Proposal` deal in `Probabilistic` instead of `Vertex`?